### DISCUSSION TITLE
How can I model my problem?

### DISCUSSION THREAD

**Discussion by Penido-exe** [neuraloperator/neuraloperator#442] [Q&A]
I'm really confused about how to model my problem:
I have 10 simulations of gas spreading through an environment. Each simulation contains 100 grids of size 30x100x4, so 100 time steps, and each frame has 4 channels.

I need to predict the state in n frames ahead in the future.

Taking a step back, an FNO seems like the right choice, right?
But I can't figure out if I should be using an FNO_2D or FNO_3D.
Can someone help me clarify that?

**Comment by JeanKossaifi** [comment]
You can do both, it all depends on the problem you are trying to solve: you can use FNO 2D if you want to go from any fixed time step to another (e.g. t -> t+1) and learn an operator over the (discretized) 2D space. FNO 3D if you want to learn a mapping between spatiotemporal blocks. Is your problem Markovian?

I'd recommend using [FNO](https://neuraloperator.github.io/dev/modules/generated/neuralop.models.FNO.html#neuralop.models.FNO) either way -- and pass either 2 or 3 values in the n_modes tuple.

**Reply by Penido-exe** [reply]
Thank you for your response.

No, my problem is not this Markovian one.

I’m trying to use the 3D FNO, but something is wrong. I’m training the network with several tensors of shape [5, 5, 113, 32], and the expected output for each one is [20, 1, 113, 32]. That is, I’m training the network so that, given the last 5 time steps (with 5 channels), it predicts the next 20 steps (with 1 channel). First, does this make sense to you? Is this possible with FNO?

Second, after training the network, I was expecting the output of the prediction to also be [20, 1, 113, 32], meaning that 20 steps( with 1 channel) in time would be predicted. However, the network is outputting a tensor of shape (1, 1, 5, 113, 32), predicting only 1 time step(with 5 channels). Why is this happening?
